tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -89,9 +103,11 @@ def __init__(self):
 
         self._tfs_enable_batching = SAGEMAKER_BATCHING_ENABLED == "true"
         self._tfs_default_model_name = os.environ.get("TFS_DEFAULT_MODEL_NAME", "None")
-        self._tfs_wait_time_seconds = int(os.environ.get("SAGEMAKER_TFS_WAIT_TIME_SECONDS", 300))
         self._tfs_inter_op_parallelism = os.environ.get("SAGEMAKER_TFS_INTER_OP_PARALLELISM", 0)
         self._tfs_intra_op_parallelism = os.environ.get("SAGEMAKER_TFS_INTRA_OP_PARALLELISM", 0)
+        self._tfs_instance_count = int(os.environ.get("SAGEMAKER_TFS_INSTANCE_COUNT", 1))
+        self._tfs_wait_time_seconds = int(
+            os.environ.get("SAGEMAKER_TFS_WAIT_TIME_SECONDS", 55 // self._tfs_instance_count))
```

Comment:
When customer sends invoke endpoint request. Sagemaker will make following calls to TFS DLC:
1. Load model request. ( this request will load `SAGEMAKER_TFS_INSTANCE_COUNT` number of tfs instance )
2. Invoke endpoint.
The total timeout is 60s.

Since we are starting tfs instance one by one, we give each tfs 55/tfs_count seconds to start. the rest 5s to run inference.

I am not sure why it was 300s in the first place. By given the original code `wait_for_model` (https://github.com/aws/deep-learning-containers/blob/master/tensorflow/inference/docker/build_artifacts/sagemaker/tfs_utils.py#L257) is honor timeout. I am not surprised if they just put a random number there.

Reply:
I don't love this, as it feels unnecessarily restrictive. But I guess maybe it's fine, as long as `tfs_count` doesn't get too high so that the individual timeouts are unreasonably small.

Reply:
Customer can overwrite timeout with `SAGEMAKER_TFS_WAIT_TIME_SECONDS `. But given sagemaker has hard limit (60s) on each request. I think it will timeout if customer is trying to start too many tfs instances on sagemaker endpoint.